### PR TITLE
Experiment secondary metric results UI

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

This PR adds secondary metric results to the frontend, cleans up a lot of the different experiment UI states (combos of in progress, in draft, completed, with secondary, without, blah blah), and fixes a bug with secondary metrics creation with the funnel "Add step" button.

Secondary metric results to the frontend (with feature flag on still):

Screen Shot 2022-02-14 at 10 08 45 AM

**Completed experiment without secondary metrics:**

Screen Shot 2022-02-14 at 10 09 26 AM

**In progress experiment with no results/implementation yet. feature flag info is left up here in case the user implemented it incorrectly which is why they're producing empty results**

Screen Shot 2022-02-14 at 10 09 03 AM

**Draft experiment before it's launched**

Screen Shot 2022-02-14 at 10 08 08 AM

## How did you test this code?

Create different experiments and check out their states

*Please describe.*

### CODE REVIEW

File: ee/clickhouse/queries/experiments/secondary_experiment_result.py
Comment (liyiy) [inline ee/clickhouse/queries/experiments/secondary_experiment_result.py:65]: @neilkakkar I changed the structure here because I wanted to use this to send secondary metric results back in the `experiment/results` call as one instead of making X many calls and having them load in different states on the frontend. Let me know if this makes sense and then I'll fix the backend tests accordingly for this

Comment (neilkakkar) [reply ee/clickhouse/queries/experiments/secondary_experiment_result.py:65]: Hmm, the reason they're not in one call is because this can get pretty slow: Not only are you querying for a trend/funnel breakdown, but also the significance calculation.

Add to that a few more result calls, and you become 3-5x slower. I think it's fine to break this down, and have small spinners for each of the secondary metrics, that load later than the main big one.

Thoughts?

Comment (neilkakkar) [reply ee/clickhouse/queries/experiments/secondary_experiment_result.py:65]: Also, I'd guess secondary metrics only come into play when results are significant and you're making a final decision. So, main metric should always have the priority + major screen time.


(Will take a look at this PR in a bit!)

Comment (liyiy) [reply ee/clickhouse/queries/experiments/secondary_experiment_result.py:65]: Alrighty I'll redo the way the frontend loads the results

File: frontend/src/scenes/experiments/Experiment.tsx
Comment (neilkakkar) [inline frontend/src/scenes/experiments/Experiment.tsx:936]: instead of the spinner, default state now seems to be the text "Loading Data...", which isn't great.

Comment (neilkakkar) [reply frontend/src/scenes/experiments/Experiment.tsx:936]: (try throttling on chrome)

Comment (neilkakkar) [reply frontend/src/scenes/experiments/Experiment.tsx:936]: This affects non-FF'ed people, so think we should fix this before merging

**Comment by liyiy** [comment]
ALRIGHT! This is good to go now I think, some UI improvements can continue to be iterated on but not in this PR maybe lol

**Review by neilkakkar — APPROVED** [review]
LGTM.

Let me know if the change makes sense: this ensures we're not waiting for the results to load before we show the experiment header.

**Comment by liyiy** [comment]
> LGTM.
> 
> Let me know if the change makes sense: this ensures we're not waiting for the results to load before we show the experiment header.

No I changed it to loading because experimentData will always render something since it holds previous experiment values because the logic doesn't unmount

**Comment by neilkakkar** [comment]
Ha, interesting! I guess that's true for results as well?

In that case, probably makes sense to clear experiment results on `loadExperiment`, and bring back `experimentLoading`, but not `experimentResultsLoading`.

**Pull request merged**
